lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -317,43 +309,39 @@ withNetworkLayerBase tr np conn (versionData, _) action = do
     -- doesn't rely on the intersection to be up-to-date.
     let handlers = retryOnConnectionLost tr
 
-    queryRewardQ <- connectDelegationRewardsClient handlers
-
-    (nodeTipChan, networkParamsVar, interpreterVar, localTxSubmissionQ) <-
+    (readNodeEraAndTip, networkParamsVar, interpreterVar, localTxSubmissionQ) <-
         connectNodeTipClient handlers
 
+    -- Alternative read-action which waits for the era to be Just and uses Point
+    -- instead of Tip.
+    let readNodeEraAndTip' = readNodeEraAndTip >>= \case
```

Comment:
Could possibly converge on some neat api of `STM` actions for node-tip state, but leaving it like this for now.

Reply:
On a small stylistic note, this logic is quite specific to the `currentNodeEra` handler so I'd simply leave in there rather than defining a top-level let binding.